python/tests/core/test_declarative_schema.py

```diff
@@ -117,40 +143,42 @@ def test_enabled_metrics(fi_disabled: bool, unicode_enabled: bool) -> None:
     assert unicode_enabled == ("unicode_range" in col1_metrics)
 
 
-@pytest.mark.parametrize(
-    "spec,msg",
-    [
-        (ResolverSpec(), "DeclarativeSchema: resolver specification must supply name or type"),
-        (ResolverSpec(column_type=str), "DeclarativeSchema: resolver specification column type must be a DataType"),
-        (
-            ResolverSpec(column_name="bruce", metrics=[MetricSpec(int)]),
-            "DeclarativeSchema: must supply a Metric subclass to MetricSpec",
-        ),
-    ],
-)
-def test_invalid_config(spec, msg) -> None:
+def test_invalid_config() -> None:
     """
     Verify error checking
     """
-    with pytest.raises(ValueError, match=msg):
-        DeclarativeResolver([spec])
-
-
-def test_standard_resolver() -> None:
+    with pytest.raises(ValueError) as e:
+        ResolverSpec()
+        assert e.value.args[0] == "ResolverSpec: resolver specification must supply name or type"
+    with pytest.raises(ValueError) as e:
+        ResolverSpec(column_type=str)
+        assert e.value.args[0] == "ResolverSpec: resolver specification column type must be a DataType"
+    with pytest.raises(ValueError) as e:
+        ResolverSpec(column_name="bruce", metrics=[MetricSpec(int)])
+        assert e.value.args[0] == "MetricSpec: must supply a Metric subclass to MetricSpec"
+
+
+def test_resolvers() -> None:
     """
-    Verify DeclarativeSchema(STANDARD_RESOLVER) is equivalent to DatasetSchema()
+    Verify DeclarativeSchema(RESOLVER) is equivalent to DatasetSchema()
+    for different RESOLVERS (Standard, Limited Tracking, HistogramCounting)
     """
+    params = [
```

Comment:
Could make this a `@pytest.mark.parametrize`

Reply:
I tried, but pytest was not happy about it, for some reason.
that was a workaround, but hopefully an acceptable one